src/TesApi.Web/BatchScheduler.cs

```diff
@@ -573,6 +526,70 @@ static bool IsPoolQuotaException(string code)
                     "PoolQuotaReached" => true,
                     _ => false,
                 };
+
+            void HandleException(Exception exception)
```

Comment:
This error handling seems complex to me. At the very least we should have a test that validates the desired behavior.

Reply:
I will add one